olive/engine/cloud_cache_helper.py

```diff
@@ -0,0 +1,244 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import io
+import json
+import logging
+import os
+from copy import deepcopy
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict
+
+from olive.common.config_utils import ConfigBase
+from olive.common.utils import get_credentials, hash_dict
+from olive.engine.config import FAILED_CONFIG, PRUNED_CONFIGS
+from olive.model.config.model_config import ModelConfig
+
+logger = logging.getLogger(__name__)
+
+
+class CloudCacheConfig(ConfigBase):
+    enable_cloud_cache: bool = True
+    account_url: str = "https://olivepublicmodels.blob.core.windows.net"
```

Comment:
Should we remove this and keep it as required by default?

Reply:
No, the common use case is all users will share a same model storage account. I make this configurable just in case there is any custom use case.